From feac54227936507d189d0ec3a00b076baece811b Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Wed, 27 Nov 2013 11:34:56 +0000
Subject: [PATCH 295/351] ASoC: wm_adsp: Remove work around for firmware bug

We no longer need to avoid fully emptying the circular buffer on the
DSP.

Change-Id: I0c6eda75d1920c0c66e9ee9150ddd8771debbec3
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
---
 sound/soc/codecs/wm_adsp.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/sound/soc/codecs/wm_adsp.c b/sound/soc/codecs/wm_adsp.c
index 6e48307..7f5ed3b 100644
--- a/sound/soc/codecs/wm_adsp.c
+++ b/sound/soc/codecs/wm_adsp.c
@@ -2653,9 +2653,6 @@ static int wm_adsp_capture_block(struct wm_adsp *adsp, int *avail)
 	read_index = sign_extend32(next_read_index, 23);
 	write_index = sign_extend32(next_write_index, 23);
 
-	/* Don't empty the buffer as it kills the firmware */
-	write_index--;
-
 	if (read_index < 0)
 		return -EIO;	/* stream has not yet started */
 
-- 
1.7.9.5

